src/otx/v2/adapters/torch/mmengine/mmdet/engine.py

```diff
@@ -0,0 +1,214 @@
+                string representing the path to the image file, a NumPy array containing the image data, or a list of
+                NumPy arrays containing multiple images. Defaults to None.
+            checkpoint (Optional[Union[str, Path]], optional): The path to the checkpoint file to use for inference.
+                Defaults to None.
+            pipeline (Optional[Union[Dict, List]], optional): The data pipeline to use for inference. Can be a
+                dictionary containing the pipeline configuration, or a list of dictionaries containing multiple
+                pipeline configurations. Defaults to None.
+            device (Union[str, torch.device, None], optional): The device to use for inference. Can be a string
+                representing the device name (e.g. 'cpu' or 'cuda'), a PyTorch device object, or None to use the
+                default device. Defaults to None.
+            task (Optional[str], optional): The type of task to perform. Defaults to None.
+            batch_size (int, optional): The batch size to use for inference. Defaults to 1.
+            **kwargs: Additional keyword arguments to pass to the inference function.
+
+        Returns:
+            List[Dict]: A list of dictionaries containing the inference results.
+        """
+        import cv2
+        from mmcv.transforms import Compose
+        from mmdet.apis import DetInferencer, inference_detector
+        from mmengine.model import BaseModel
+
+        # Model config need data_pipeline of test_dataloader
+        # Update pipelines
+        if pipeline is None:
+            pipeline = [
+                {"type": "Resize", "scale": [512, 512]},
+                {"type": "PackDetInputs", "meta_keys": ['img', 'img_id', 'scale', 'img_shape', 'scale_factor']},
+            ]
+
+        cfg = Config({})
+        if isinstance(model, torch.nn.Module) and hasattr(model, "cfg"):
+            cfg = model.cfg
+        elif isinstance(model, dict) and "cfg" in model:
+            cfg = model["cfg"]
+        cfg["test_dataloader"] = {"dataset": {"pipeline": pipeline}}
+        if isinstance(model, dict):
+            model.setdefault("cfg", cfg)
+        elif isinstance(model, torch.nn.Module):
+            model.cfg = cfg
+
+        # Check if the model can use mmdet's inference api.
+        if isinstance(checkpoint, Path):
+            checkpoint = str(checkpoint)
+        if isinstance(model, BaseModel):
+            if isinstance(img, str):
+                img = cv2.cvtColor(cv2.imread(img), cv2.COLOR_BGR2RGB)
+            inputs = {
+                "model": model,
+                "imgs": img,
+                "test_pipeline": Compose(pipeline),
+            }
+            return [inference_detector(**inputs, **kwargs)]
+        inferencer = DetInferencer(
+            model=model,
+            weights=checkpoint,
+            device=device,
+        )
+
+        return inferencer(img, batch_size, **kwargs)
```

Comment:
Based on this rule
https://docs.astral.sh/ruff/rules/unnecessary-assign/

```suggestion
        return DetInferencer(model=model, weights=checkpoint, device=device)
```

Reply:
I don't agree with this suggestion. It should be return DetInferencer(model=model, weights=checkpoint, device=device)(img, batch_size, **kwargs), but I don't think is clean implementation.

Reply:
I also agree with Jaeguk's opinion.